Sources/StreamChatUI/Utils/ImageLoading/DefaultImageLoader.swift

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Nuke
+import UIKit
+
+extension ImageTask: Cancellable {}
+
+/// The class which is resposible for loading images from URLs.
+/// Internally uses `Nuke`'s shared object of `ImagePipeline` to load the image.
+open class DefaultImageLoader: ImageLoading {
+    /// The CDN to be used
+    open var imageCDN: ImageCDN
+    
+    public init(imageCDN: ImageCDN) {
```

Comment:
weird coupling here

Reply:
Agreed, image loading shouldn't have any knowledge about CDN. Changed the code to remove this coupling